lightning/src/sign/mod.rs

```diff
@@ -280,13 +326,22 @@ impl SpendableOutputDescriptor {
 			match outp {
 				SpendableOutputDescriptor::StaticPaymentOutput(descriptor) => {
 					if !output_set.insert(descriptor.outpoint) { return Err(()); }
+					let sequence =
+						if descriptor.channel_transaction_parameters.as_ref()
+							.map(|channel_params| channel_params.channel_type_features.supports_anchors_zero_fee_htlc_tx())
+							.unwrap_or(false)
+						{
+							Sequence::from_consensus(1)
+						} else {
+							Sequence::ZERO
+						};
 					input.push(TxIn {
 						previous_output: descriptor.outpoint.into_bitcoin_outpoint(),
 						script_sig: Script::new(),
-						sequence: Sequence::ZERO,
+						sequence,
 						witness: Witness::new(),
 					});
-					witness_weight += StaticPaymentOutputDescriptor::MAX_WITNESS_LENGTH;
+					witness_weight += descriptor.max_witness_length();
```

Comment:
Hmmm... I'd definitely lean towards a separate variant, FWIW. Seems cleaner.

Reply:
I see. My thinking was that by explicitly enumerating the cases, we would be forced to consider all of them and avoid bugs. I suppose an alternative would be to have a mapping from `ChannelTypeFeatures` to an enum with a variant for each type that we could match on.

Reply:
SGTM. A mapping wouldn't require any serialization changes, too.

Reply:
Just to expand on this based on offline discussion. We'll keep the variants as they are in this PR. This will give us a `ChannelTransactionParameters` which has a `ChannelTypeFeatures`. Later, we can add something like a `to_channel_type` method to `ChannelTypeFeatures` for internal use that will give an enum for us to match so as to avoid missing cases.

Using either another variant or a nested enum instead could be error prone as it may go out of sync with `ChannelTypeFeatures`.